fix(markdown widget): adds keyboard shortcuts (#3005)

kwyoung11 opened the pull request:
adds keyboard shortcuts for bold, code, italic and strikethrough marks, inline links and block headers.

<!--
Thanks for submitting a pull request!

Please make sure you've read and understood our contributing guidelines.

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx", where #xxxx is the issue number.

Please provide enough information so that others can review your pull request.
The first two fields are mandatory:
-->

**Summary**
adds keyboard shortcuts to markdown widget:
'mod+b': bold
'mod+shift+c': code
'mod+i': italic
'mod+shift+s': strikethrough
'mod+k': link
'mod+1': h1
'mod+2': h2
'mod+3': h3
'mod+4': h4
'mod+5': h5
'mod+6': h6
  
<!--
Explain the **motivation** for making this change.
What existing problem does the pull request solve?
-->

**Test plan**
Added Cypress tests to test each keyboard shortcut
<img width="716" alt="Screen Shot 2020-04-13 at 2 52 26 AM" src="https://user-images.githubusercontent.com/1297966/79103308-f07fd380-7d31-11ea-8526-1900dd87f27a.png">

erezrokah commented:
Thanks @kwyoung11! Linking https://github.com/netlify/netlify-cms/issues/3005 for context

kwyoung11 commented:
@erezrokah sure thing. I had to change the heading hotkeys to just mod+{number} from mod+shift+{number} because on Mac OS, mod+shift+3, mod+shift+4 and mod+shift+5 are shortcuts already defined by the OS. If any of the shortcuts are non-standard though, let me know. Going to pick up testing tomorrow.

kwyoung11 commented:
- Added Cypress tests in `markdown_widget_hotkeys_spec.js`
- Added a HOT_KEY_MAP constant so we have the key maps defined in one place.

Unfortunately I had to duplicate the HOT_KEY_MAP from plugins/Hotkey into cypress/utils/constants since Cypress does not support ES6 syntax and I was getting the error `Cypress ParseError: 'import' and 'export' may appear only with 'sourceType: module'` when trying to import the HOT_KEY_MAP into the test file. Would it be worth considering adding webpack as a pre-processor to Cypress in a separate PR so that this duplication would not be necessary? (See https://github.com/cypress-io/cypress-webpack-preprocessor and https://stackoverflow.com/a/56693706/1205460 for more on this).

Also, Slate wraps inline elements, e.g. links, with BOM characters, so I needed to replace those in the `confirmMarkdownEditorContent` method in support/commands.js.
